Update code_style.md (#12)
unfoldtoolbox/UnfoldDocs: src/code_style.md

2. Try to structure your program into meaningful subfunctions. As a rule of thumb, if your function is longer than 50 lines of code, has nested loops, nested if-else clauses, you'd better have a justification for it, or refactor it into smaller, reusable (and testable) functions.
3. Structure your functions into multiple files, separated on a conceptual basis.
4. Try to keep [formatting consistent across all files](https://unfoldtoolbox.github.io/UnfoldDocs/main/contr_formatting/).
5. Minimize external dependencies. Use external libraries only when necessary to avoid compatibility issues and reduce installation and load times.
6. Comment your code, especially if the code is not self-explanatory.
7. Modularity: It's best if users can customize functionality without having to modify the package code. Good practices: avoid hardcoded parameters, make attributes (e.g. in figures) customizable, use inheritance from `AbstractTypes`.

> 6. Most users will not check the defaults, make sure they are sensible and nudge users to provide their own.

Reviewer: they - users or defaults?

Author: hahaha both :D
```suggestion
6. Most users will not check the defaults, make sure the defaults are sensible, potentially nudge users to provide their own with warnings.
```

# Backward compatibility 

> Idally, new versions should produce the same results without any changes to the user's code. If not, a breaking release is required to indicate changed behavior.

Reviewer: Ideally

Author: ```suggestion
Ideally, new versions should produce the same results without any changes to the user's code. If not, a breaking release is required to indicate changed behavior.
```
